lib/ws_servers/api/send_recurring_algo_order_list.js

```diff
@@ -6,17 +6,36 @@ const sendError = require('../../util/ws/send_error')
 const mapRecurringAlgoOrderState = require('../../util/map_recurring_ao_state')
 const restPaginationHandler = require('../../util/rest_pagination_handler')
 
+const getRecurringAosWithRetry = async (rest, debug) => {
+  let algoOrders = []
+  let retries = 0
+  let receivedData = false
+  let err = null
+  while (!receivedData && retries < 3) {
+    try {
+      algoOrders = await restPaginationHandler(rest.getRecurringAlgoOrders.bind(rest), {})
+      receivedData = true
+      err = null
+      debug('fetched total Recurring AOs', algoOrders.length)
+    } catch (e) {
+      debug('error fetching Recurring AOs', e)
+      err = e
+      retries++
+    }
+  }
+  if (err) throw err
+
+  return algoOrders
+}
+
 /**
  * Get all recurring algo order list
  *
  * @param {RESTv2} rest
  * @returns {Promise} p
  */
-const getAndUpdateRecurringAlgoOrdersList = async (rest, algoDB) => {
-  const algoOrders = await restPaginationHandler(
-    rest.getRecurringAlgoOrders.bind(rest),
-    {}
-  )
+const getAndUpdateRecurringAlgoOrdersList = async (rest, algoDB, debug) => {
```

Comment:
new args not included in the jsdoc above

Reply:
Fixed